resources/splunk_app.rb

```diff
@@ -136,3 +148,41 @@ def config_scope
     apply_config
   end
 end
+
+class PackagedApp < SplunkApp
+  property :source_url, String, required: true
+
+  resource_name :splunk_app_package
+
+  def package_path
+    return @package_path if @package_path
+    file_cache = Pathname.new(Chef::Config['file_cache_path'])
+    @package_path = file_cache.join(CernerSplunk::PathHelpers.filename_from_url(source_url).gsub(/.spl$/, '.tgz'))
+  end
+
+  action :install do
+    return unless !version || changed?(:version)
+
+    remote_file package_path.to_s do
```

Comment:
After offline discussion, I've learned that we want to validate that, if the desired version and installed version share the same release version, the desired version must not be a prerelease version.

This will require some parsing, and likely encapsulation in a new object.